src/components/trace/TraceEventBlock.vue

```diff
@@ -120,6 +120,13 @@ export default {
       height: 0,
     };
   },
+  watch: {
+    '$store.state.focusedEvent': {
+      handler() {
+        this.initialize();
+      },
+    },
+  },
```

Comment:
I'd consider adding this as a prop instead of adding it to the global store. In the past, repurposing components which have a hard dependency on the store has been difficult.

Reply:
I made it with `watch` because I thought it's not necessary to pass a prop through long chain:
VsCodeExtention -> DiagramTrace -> Trace -> TraceEventBlock

Now focused event is passing as prop: https://github.com/applandinc/appmap-js/pull/168/commits/87b4aa91f0c1f4f00155bb050d2da0dfb0838de2

Reply:
In the future, I think we'll end up needing to add this state directly to the `Event` object. I agree, propagating props through many components (they call this prop drilling) feels a bit odd, but it ends up keeping the components more clean as they don't rely on global state.